activerecord/lib/active_record/scoping/default.rb

```diff
@@ -115,7 +115,8 @@ def build_default_scope(base_rel = nil) # :nodoc:
             base_rel ||= relation
             evaluate_default_scope do
               default_scopes.inject(base_rel) do |default_scope, scope|
-                default_scope.merge(base_rel.scoping { scope.call })
+                scope = scope.is_a?(Proc) ? scope : scope.method(:call)
+                default_scope.merge(base_rel.scoping{instance_exec(&scope)})
```

Comment:
Should this just be `base_rel.instance_exec(&scope)`? Or can't this even be `default_scope.merge(&scope)`?